### PR TITLE
Rewrite the Caffe2 backend translator again.

### PR DESCRIPTION
**Pull request by ezyang**
I got annoyed at manually slinging protobufs around, so I
made a little wrapper type that turns attributes into a dictionary
(so much nicer!)

I also dropped support for old Caffe2 which don't know how to
handle multiple kernels/etc for ConvTranspose.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Comment by ezyang** [comment]
I ran both tests locally with a pre onnx breaking change and verified they passed.

**Comment by bddppq** [comment]
Let's get https://github.com/onnx/onnx-caffe2/pull/27 into master first so the CI can be working again

File: onnx_caffe2/backend.py [hunk 1]
```diff
@@ -120,38 +123,151 @@ def run(self, inputs, **kwargs):
...
+    We may temporarily edit these nodes to get them into Caffe2 form,
+    before actually translating into the Caffe2 protobuf, since this
+    is easier than decomposing everything, and putting it back together
+    when we're ready.
+    """
+    def __init__(self, node):
+        self.name = str(node.name)
+        self.op_type = str(node.op_type)
+        self.attrs = OnnxAttributes.from_onnx(node.attribute)
+        self.consumed_inputs = self.attrs.pop("consumed_inputs", None)
```
Comment (bddppq) [inline onnx_caffe2/backend.py]: I really appreciate this!

**Comment by bddppq** [comment]
I'm gonna resolve merge conflicts and then merge it.

**Pull request merged**